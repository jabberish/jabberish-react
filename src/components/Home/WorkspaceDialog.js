import React from 'react';
import PropTypes from 'prop-types';

import { 
  Dialog, 
  DialogTitle, 
  DialogContent, 
  DialogActions, 
  DialogContentText, 
  Button, 
  TextField } from '@material-ui/core';

const WorkspaceDialog = ({ open, handleCloseDialog, handleCreateWorkspace, handleUpdate }) => {
  return (
    <Dialog 
      open={open} 
      onClose={handleCloseDialog} 
      aria-labelledby="form-dialog-title"
    >
      <DialogTitle id="form-dialog-title">Create Workspace</DialogTitle>
      <DialogContent>
        <DialogContentText>
            Please provide a name for your new workspace.
        </DialogContentText>
        <TextField
          onChange={handleUpdate}
          autoFocus
          margin="dense"
          id="newWorkspaceName"
          name="newWorkspaceName"
          label="Workspace name"
          type="text"
          fullWidth
        />
      </DialogContent>
      <DialogActions>
        <Button onClick={handleCloseDialog} color="secondary">
            Cancel
        </Button>
        <Button onClick={handleCreateWorkspace} color="primary">
            Create
        </Button>
      </DialogActions>
    </Dialog>
  );
};

WorkspaceDialog.propTypes = {
  open: PropTypes.bool.isRequired,
  handleCloseDialog: PropTypes.func.isRequired,
  handleCreateWorkspace: PropTypes.func.isRequired,
  handleUpdate: PropTypes.func.isRequired
};

export default WorkspaceDialog;
