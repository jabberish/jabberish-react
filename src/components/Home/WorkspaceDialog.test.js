import React from 'react';
import { shallow } from 'enzyme';
import WorkspaceDialog from './WorkspaceDialog';

describe('WorkspaceDialog component', () => {
  it('renders WorkspaceDialog', () => {
    const wrapper = shallow(<WorkspaceDialog 
      open={true} 
      handleCloseDialog={() => {}} 
      handleCreateWorkspace={() => {}}
      handleUpdate={() => {}}
    />);
    expect(wrapper).toMatchSnapshot();
  });
});
  
