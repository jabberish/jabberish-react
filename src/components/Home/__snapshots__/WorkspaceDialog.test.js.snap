// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WorkspaceDialog component renders WorkspaceDialog 1`] = `
<WithStyles(ForwardRef(Dialog))
  aria-labelledby="form-dialog-title"
  onClose={[Function]}
  open={true}
>
  <WithStyles(ForwardRef(DialogTitle))
    id="form-dialog-title"
  >
    Create Workspace
  </WithStyles(ForwardRef(DialogTitle))>
  <WithStyles(ForwardRef(DialogContent))>
    <WithStyles(ForwardRef(DialogContentText))>
      Please provide a name for your new workspace.
    </WithStyles(ForwardRef(DialogContentText))>
    <WithStyles(ForwardRef(TextField))
      autoFocus={true}
      fullWidth={true}
      id="newWorkspaceName"
      label="Workspace name"
      margin="dense"
      name="newWorkspaceName"
      onChange={[Function]}
      type="text"
    />
  </WithStyles(ForwardRef(DialogContent))>
  <WithStyles(ForwardRef(DialogActions))>
    <WithStyles(ForwardRef(Button))
      color="secondary"
      onClick={[Function]}
    >
      Cancel
    </WithStyles(ForwardRef(Button))>
    <WithStyles(ForwardRef(Button))
      color="primary"
      onClick={[Function]}
    >
      Create
    </WithStyles(ForwardRef(Button))>
  </WithStyles(ForwardRef(DialogActions))>
</WithStyles(ForwardRef(Dialog))>
`;
